api/v1alpha1/nodepool_types.go

```diff
@@ -413,11 +413,67 @@ type PowerVSNodePoolPlatform struct {
 	ImageDeletePolicy string `json:"imageDeletePolicy,omitempty"`
 }
 
+// KubevirtCompute contains values associated with the virtual compute hardware requested for the VM.
+type KubevirtCompute struct {
+	// Memory represents how much guest memory the VM should have
+	//
+	// +optional
```

Comment:
I assumed some of these optional values are necessary for the controller to work. Why arent' we defaulting values here and documenting it? so as a consumer when I look at the API docs I'll see something like // Memory represents how many cores the guest VM should have. If unset the default value is x

Reply:
i added kubebuilder defaults and better documented the defaults